casper/src/main/scala/coop/rchain/casper/SafetyOracle.scala

```diff
@@ -134,8 +146,9 @@ sealed abstract class SafetyOracleInstances {
           candidateMetadata: BlockMetadata
       ): F[Map[BlockHash, Long]] =
         candidateMetadata.parents.headOption match {
-          case Some(parent) => blockDag.lookup(parent).map(_.get.weightMap)
-          case None         => candidateMetadata.weightMap.pure[F]
+          case Some(parent) =>
+            blockDag.lookup(parent).map(_.fold(candidateMetadata.weightMap)(_.weightMap))
+          case None => candidateMetadata.weightMap.pure[F]
```

Comment:
You can make this little nicer without duplicating default value. With _traverse_ (_sequence_) you can "invert" two monads (`F` can be applicative only) `Option[F[A]]` to `F[Option[A]]`.
Also we can save the creation of one Task. In runtime `F` will be a Monix Task, so wrapping a value with `pure[F]` will create a new Task just to immediately return this value. We should investigate how big is this overhead especially in large loops.
```scala
import cats.instances.option._
for {
  maybeMetadata <- candidateMetadata.parents.headOption.flatTraverse(blockDag.lookup)
} yield maybeMetadata.map(_.weightMap).getOrElse(candidateMetadata.weightMap)
```